x-pack/plugins/observability/public/pages/slo_details/slo_details.tsx

```diff
@@ -53,6 +57,31 @@ export function SloDetailsPage() {
 
   useBreadcrumbs(getBreadcrumbs(basePath, slo));
 
+  useEffect(() => {
+    if (!slo) {
+      return;
+    }
+
+    return setApplicationContext({
+      description: dedent(`
+        The user is looking at the detail page for the following SLO
+        
+        Name: ${slo.name}.
+        Id: ${slo.id}
+        Description: ${slo.description}
+        Observed value: ${slo.summary.sliValue}
+        Status: ${slo.summary.status}
+      `),
+      data: [
+        {
+          name: 'slo',
+          description: 'The SLO and its metadata',
+          value: slo,
+        },
+      ],
+    });
```

Comment:
I'm not convinced having a structured context is better. What is the difference from the above and a string context? Do we have use cases for overwriting values?

```suggestion
return setApplicationContext(dedent(`
  The user is looking at the detail page for the following SLO
  
  Name: ${slo.name}.
  Id: ${slo.id}
  Description: ${slo.description}
  Observed value: ${slo.summary.sliValue}
  Status: ${slo.summary.status}
`));
```

As a consumer I am wondering what the difference is between putting something in the top-level description or the `data[0].description`. It is also not clear whether `name` affects the prompt.

Reply:
I think in some cases you'll want natural language, but I don't think it's necessary for everything. E.g. in this case, the `slo` object has lots of interesting data, I don't want to write that out in natural language. And the LLM is pretty good at understanding structured data.

Reply:
>  I don't want to write that out in natural language. And the LLM is pretty good at understanding structured data.

Sure, but you could output it as `${JSON.stringify(slo)}` inside the string. In my example above it looked very redundant because you already output it in "natural language" as you say:

```
return setApplicationContext(dedent(`
  The user is looking at the detail page for the following SLO
  
  Name: ${slo.name}.
  Id: ${slo.id}
  Description: ${slo.description}
  Observed value: ${slo.summary.sliValue}
  Status: ${slo.summary.status}

  ${JSON.stringify(slo)}
`));
```